xarray/core/computation.py

```diff
@@ -408,7 +434,20 @@ def apply_dict_of_variables_vfunc(
 
     result_vars = {}
     for name, variable_args in zip(names, grouped_by_name):
-        result_vars[name] = func(*variable_args)
+        core_dim_check = _check_core_dims(signature, variable_args, name)
+        if core_dim_check is True:
+            result_vars[name] = func(*variable_args)
+        else:
+            if missing_core_dim == "raise":
+                raise ValueError(core_dim_check)
+            elif missing_core_dim == "copy":
+                result_vars[name] = variable_args[0]
+            elif missing_core_dim == "drop":
+                pass
+            else:
+                raise ValueError(
```

Comment:
We should check to allow only`"raise"` is the only valid option for dataarrays as input.

Reply:
Actually this code path only gets hit for Datasets — do we want to pre-emptively check for the kwarg when there are only DataArrays?

Reply:
Yes I think so.

Reply:
I'm in gentle disagreement — if someone passes `apply_ufunc(f, da, on_missing_core_dims="copy")` then the claim is that we want to raise immediately?

I think that's going to lead to lots of callers having to check the type and choose what to pass based on whether it's a `da` vs. `ds`. The motivating example from https://github.com/pydata/xarray/pull/8114 would need to do this.

Instead, surely it's consistent to allow `drop` & `copy` for arrays, but know they'll get ignored?

Reply:
> think that's going to lead to lots of callers having to check the type and choose what to pass based on whether it's a da vs. ds.

Ok good point. I'm convinced!